shadows/framework/src/main/java/org/robolectric/shadows/ShadowBiometricManager.java

```diff
@@ -65,9 +109,37 @@ public void setCanAuthenticate(boolean flag) {
     biometricServiceConnected = flag;
   }
 
-  @Implementation(minSdk = R)
-  protected int canAuthenticate(int userId, int authenticators) {
-    return biometricServiceConnected ? BIOMETRIC_SUCCESS : BIOMETRIC_ERROR_NO_HARDWARE;
+  /**
+   * @param num to set the number of enrolled biometric. If sets the value equal or below 0, result
+   *     will be {@link BiometricManager#BIOMETRIC_ERROR_NONE_ENROLLED}
+   */
+  public void setEnrolledBiometrics(int num) {
+    enrolledBiometrics = num;
+  }
+
+  /**
+   * @param flag to set has biometric hardware or not. If sets the value false, result will be
+   *     {@link BiometricManager#BIOMETRIC_ERROR_HW_UNAVAILABLE}
+   */
+  public void hasAvailableHardware(boolean flag) {
```

Comment:
Instead of having separate methods for:
* hasAvailableHardware
* setAuthenticatorType
* requireSecurityUpdate

Why not have a single method called something like `setCanAuthenticate(result)` that just provides the return value of `canAuthenticate(int)`? I am not an expert on BiometricManager, but is there a testing use case that this doesn't support?

Reply:
@hoisie My initial thought is to have 3 methods for different use cases but your suggestion is good, having a single method could be more easier to maintain 👌

Reply:
The initial idea is from https://android-review.googlesource.com/c/platform/frameworks/support/+/2054331. If Robolectric can support all authentication result, some mock logic of this test file can be replaced by Robolectric.